### PR TITLE
tests: uninstall_by_spec error and rpath_args tests

### PR DESCRIPTION
**Pull request by tldahlgren**
These tests were originally included in PR #11797.

These test cases are intended to increase Spack code coverage.

### CODE REVIEW

File: lib/spack/spack/test/install.py [hunk 1]
```diff
@@ -275,6 +276,25 @@ class MockInstallError(spack.error.SpackError):
...
 
+def test_uninstall_by_spec_errors(mutable_database):
+    """Test exceptional cases with the uninstall command."""
+
+    # Try to uninstall a spec that has not been installed
+    rec = mutable_database.get_record('zmpi')
+    with pytest.raises(InstallError, matches="not installed"):
+        PackageBase.uninstall_by_spec(rec.spec)
+
+    # Try uninstall -- without forcing -- a spec with dependencies
```
Comment (becker33) [inline lib/spack/spack/test/install.py]: This comment should read `...a spec with dependents`

File: lib/spack/spack/test/install.py [hunk 2]
```diff
@@ -275,6 +276,25 @@ class MockInstallError(spack.error.SpackError):
...
+    rec = mutable_database.get_record('mpich')
+
+    rpath_args = rec.spec.package.rpath_args
+    assert rpath_args.find('-rpath') > 0
+    assert rpath_args.rfind('mpich') > 0
```
Comment (becker33) [inline lib/spack/spack/test/install.py]: What are these assertions checking for?

Comment (tldahlgren) [reply lib/spack/spack/test/install.py]: Last-minute code coverage addition that should've been added to their own test.

Comment (tldahlgren) [reply lib/spack/spack/test/install.py]: Moved the coverage tests to separate case.

**Pull request merged**